src/main/java/io/getunleash/DefaultUnleash.java

```diff
@@ -154,6 +164,19 @@ public Variant getVariant(String toggleName, UnleashContext context, Variant def
         return variant;
     }
 
+    private Variant getVariantForMetrics(String toggleName, UnleashContext context) {
+        UnleashContext enhancedContext = context.applyStaticFields(config);
+        return resolveVariant(toggleName, enhancedContext, Variant.DISABLED_VARIANT);
+    }
+
+    private Variant resolveVariant(
+            String toggleName, UnleashContext enhancedContext, Variant defaultValue) {
+        Optional<FlatResponse<VariantDef>> response =
+                Optional.ofNullable(this.featureRepository.getVariant(toggleName, enhancedContext));
+        Optional<VariantDef> variantDef = response.map(r -> r.value);
+        return YggdrasilAdapters.adapt(variantDef, defaultValue);
+    }
```

Comment:
We need private resolve variant method that does not dispatch impression events.